daliuge-engine/dlg/manager/cmdline.py

```diff
@@ -89,7 +89,7 @@ def addCommonOptions(parser, defaultPort):
     parser.add_option(      "--cwd", action="store_true",
                       dest="cwd", help="Short for '-w .'", default=False)
     parser.add_option("-w", "--work-dir",
-                      help="Working directory, defaults to '/' in daemon mode, '.' in interactive mode", default=None)
+                      help="Working directory, defaults to '/' in daemon mode, '.' in interactive mode", default=utils.getDlgWorkDir())
```

Comment:
When the default gets picked, we need to ensure it actually exists; otherwise the NM fails to start. This is probably what's causing the unit tests failures in Travis.

Reply:
I would thus rewrite this: I'd keep the `None` as the default value, and then react to this value being `None` to actually create the directory and change to it.